### PR TITLE
LG-3797: Make Popover props optional in Menu

### PR DESCRIPTION
**Pull request by spark33**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3797](https://jira.mongodb.org/browse/LG-3797)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

**Review by shaneeza — COMMENTED** [review]
Can you add some tests in components that use Popover?

**Comment by spark33** [comment]
> Can you add some tests in components that use Popover?

@shaneeza What would we be testing? These props aren't required, so the appropriate tests would be to make sure passing these props won't cause TS errors, and neither will NOT passing these props, and that should be handled by all instances of `render` inside all tests of components that use Popover, no?

File: package.json [hunk 1]
```diff
@@ -43,7 +43,6 @@
     "@types/lodash": "^4.14.170",
     "@types/react": "18.2.23",
     "@types/react-dom": "18.2.8",
-    "@types/react-transition-group": "4.4.8",
```
Comment (shaneeza) [inline package.json:46]: why are you removing this? is it because its a devDep?

Comment (TheSonOfThomp) [reply package.json:46]: We're also installing this within individual packages, and having it here could cause some version discrepancies

**Comment by TheSonOfThomp** [comment]
@spark33 Was playing around with dependencies, and it looks like adding "@types/react-transition-group": "^4.4.8",  to `Popover` dependencies (not dev) should resolve the issue

**Comment by TheSonOfThomp** [comment]
Closing in favor of https://github.com/mongodb/leafygreen-ui/pull/2159

**Pull request merged**